### DISCUSSION TITLE
Variables Definition

### DISCUSSION THREAD

**Discussion by p-rapan** [thelfer/tfel#350] [General]
Hello,

Just a simple question. Is there any way to know the definition of the state variables (such as the p - equivalent plastic strain) ? 

Or even the definition for equivalent stresses. 

Thanks a lot!

**Comment by thelfer** [comment]
Well, those definitions are left to the user, except if you use the `StandardElastoViscoplasticity` brick or specialised DSL (such as `IsotropicMisesCreep`, etc..). See: https://thelfer.github.io/tfel/web/StandardElastoViscoPlasticityBrick.html

**Reply by p-rapan** [reply]
I am referring to the implemented Drucker-Prager and Cap model. I still find it a bit difficult to extract the expression of the equivalent plastic strain for the DP model.

**Reply by Asocie13** [reply]
There is not direct expression of the equivalent plastic strain for the DPC model. The equivalent plastic strain of the Drucker-Prager part is obtained by solving the nonlinear system with the implicit solver.  If you want to obtain by another way the Druker-Prager plasticity, I recommend you to work on the Druker-Prager models behaviors that you will find in the folder mfront/tests/behaviours

**Reply by thelfer** [reply]
@Padrp, @Asocie13 In most cases, the equivalent plastic strain is identified to the integral of the plastic multiplier.
